generate/config.go

```diff
@@ -51,6 +51,16 @@ type TypeBinding struct {
 	Unmarshaler       string `yaml:"unmarshaler"`
 }
 
+// pathJoin is like filepath.Join but 1) it only takes two argsuments,
+// and b) if the second argument is an absolute path the first argument
+// is ignored (similar to how python's os.path.join() works).
+func pathJoin(a, b string) string {
```

Comment:
I would suggest a more accurate name such as `func abs(basedir, path string) string {`
This is then similar to the stdlib's `filepath.Abs` but allows you to specify the base directory instead of using the current working directory.

func [Abs](https://cs.opensource.google/go/go/+/go1.18.2:src/path/filepath/path.go;l=243) [¶](https://pkg.go.dev/path/filepath#Abs)
func Abs(path [string](https://pkg.go.dev/builtin#string)) ([string](https://pkg.go.dev/builtin#string), [error](https://pkg.go.dev/builtin#error))
Abs returns an absolute representation of path. If the path is not absolute it will be joined with the current working directory to turn it into an absolute path. The absolute path name for a given file is not guaranteed to be unique. Abs calls Clean on the result.

Reply:
`abs` isn't very accurate because there's no requirement that `a` be an absolute path.  I really see this as a "proper" path-join (and the default go implementation as not-proper).  But ymmv.

Reply:
This is an unexported helper, so you do you.

Reply:
I agree the name isn't ideal, but I don't have a better idea, so I'm fine with this. (Closest thing I can think of in the stdlib is `net/url.ResolveReference` which I don't think is the greatest name either.)